### ISSUE TITLE
Fix the Votes model

### ISSUE THREAD

**Issue by judytuna** [judytuna/shortcode-site-309#1]
Right now, the Votes relationship is incomplete. There is no way to list all of the users that voted for a particular entry. I think I missed something with reverse relationships. 

Additionally, admins should be the only ones who can see the users that voted for an entry -- that information should not be public.

**Comment by judytuna** [comment]
Added pages to view voting relationships, such as:
- /users/2/voting_for (shows you all the entries that user has voted for)
- /entries/4/voters (lists the users that voted for that particular entry)

New todo:
- make it so that only admin users can see these.
- count the votes and display that somewhere on the entry's page (again, only admins can see this). maybe another (secret, admin-only) page that orders the entries by number of votes.

**Comment by judytuna** [comment]
Having trouble with:
http://ruby.railstutorial.org/chapters/following-users#sec:a_working_follow_button_with_ajax

**Comment by judytuna** [comment]
The code to render `_cast_vote_form` was mistakenly put inside a `<table>` in `app/views/entries/show.html.erb` . 

That's why the vote/unvote buttons worked in html, but NOT when we introduced js to the mix. When it was just html, the buttons were getting rendered without `<td>` and `<tr>` inside of a `<table>` and the browser didn't care. But once we used js, js was mad that there was stuff without td and tr, so (I think) it just ignored the line about the form, which confuses me because the browser still rendered it. The html was there, but js wasn't looking at it. As a result, when I clicked on the "vote" button, it gave me weird parameters: it was giving me route stuff (action:create, controller:votes) because it was trying to go to a route because it was skipping the vote stuff. Taking the vote form out of the table totally fixed it.

HOORAY!!!!!!!!!!!!!!!!!!!!
